### ISSUE TITLE
Shutdown mechanism for ConfigurationSourceRegistry$WatcherThread

### ISSUE THREAD

**Issue by FelixJongleur42** [anotheria/configureme#12]
Similar to
https://github.com/anotheria/moskito/issues/218#issue-393135703
there should be some shutdown mechanism for ConfigurationSourceRegistry$WatcherThread

**Comment by FelixJongleur42** [comment]
This is what Tomcat tells us:
```
11-Dec-2017 12:12:03.769 WARNING [localhost-startStop-2] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [soo] appears to have started a thread named [Thread-8] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Thread.sleep(Native Method)
 org.configureme.sources.ConfigurationSourceRegistry$WatcherThread.run(ConfigurationSourceRegistry.java:164)
```

**Comment by dvayanu** [comment]
Hello Felix,
technically there is no way for a library to know if it's used in a weird reloadable context (reloadability is kindof weird for any library). 
So adding a servlet-dependency to be able to provide a start/stop listener seems a little too much for me. I would rather add the "stop-ability to the WatcherThread, but call it from MoSKito's start/stop listener instead. 
What do you think?

**Comment by dvayanu** [comment]
added shutdown() method to ConfigurationManager to be called from a Listener (net.anotheria.moskito.webui.util.StartStopListener does it already).

**Comment by dvayanu** [comment]
will be available from 2.4.0 on

**Comment by dvayanu** [comment]
fixed in 2.4.0
